frontend/lib/resolution/Resolver.cpp

```diff
@@ -874,10 +874,27 @@ QualifiedType Resolver::getTypeForDecl(const AstNode* declForErr,
     // declared type but no init, so use declared type
     typePtr = declaredType.type();
   } else if (!declaredType.hasTypePtr() && initExprType.hasTypePtr()) {
-    // init but no declared type, so use init type
-    typePtr = initExprType.type();
-    if (inferParam) {
-      paramPtr = initExprType.param();
+    bool infersCustomType = false;
+    if (auto rec = initExprType.type()->getCompositeType()) {
+      if (rec->id().isEmpty() == false) {
+        if (const auto node = parsing::idToAst(context, rec->id())) {
```

Comment:
Is there a reasonable case where a non-empty record ID would fail to give us an AST? If not, what if we make this an assertion rather than a condition? This would help reduce another level of indentation in this block of code.

Reply:
Currently a possible case in testing is for the ``string`` type, and other such types where we force it to exist whether the module is present or not. I can add a TODO to look into improving this check. I don't think it registers as "fabricated" at the moment.